src/CartesianProduct.jl

```diff
@@ -15,14 +15,17 @@ Type that represents the cartesian product.
 For the cartesian product a several sets, there exists a special
 type `CartesianProductArray`. 
 """
-struct CartesianProduct <: LazySet
-    X::LazySet
-    Y::LazySet
-    CartesianProduct(X::LazySet, Y::LazySet) = new(X, Y)
-    CartesianProduct(Xarr::Array{LazySet, 1}) = length(Xarr) == 0 ?
+struct CartesianProduct{T1<:LazySet,T2<:LazySet} <: LazySet
```

Comment:
Now you can require that the l.h.s. is, say, a `BallInf` to write specific code.
I think we currently have no application because we mainly use the array types.
But does it hurt?

Reply:
> Now you can require that the l.h.s. is, say, a BallInf to write specific code.

ok, i see the point. 

>  But does it hurt?

no at all, just i want to understand the meaning